src/components/AnnotationsModule.vue

```diff
@@ -1,42 +1,36 @@
-<script lang="ts">
-import { defineComponent } from 'vue';
-import MeasurementsRulerList from './MeasurementsRulerList.vue';
+<script setup lang="ts">
 import MeasurementsToolList from './MeasurementsToolList.vue';
 import LabelmapList from './LabelmapList.vue';
 import ToolControls from './ToolControls.vue';
 import { usePolygonStore } from '../store/tools/polygons';
 import { useRectangleStore } from '../store/tools/rectangles';
+import { useRulerStore } from '../store/tools/rulers';
+import { AnnotationToolStore } from '../store/tools/useAnnotationTool';
+import MeasurementRulerDetails from './MeasurementRulerDetails.vue';
 
-export default defineComponent({
-  components: {
-    MeasurementsRulerList,
-    MeasurementsToolList,
-    LabelmapList,
-    ToolControls,
+const tools = [
+  {
+    store: useRulerStore(),
+    icon: 'mdi-ruler',
+    details: MeasurementRulerDetails,
   },
-  setup() {
-    return {
-      rectangleStore: useRectangleStore(),
-      polygonStore: usePolygonStore(),
-    };
+  {
+    store: useRectangleStore() as unknown as AnnotationToolStore<string>,
+    icon: 'mdi-vector-square',
   },
-});
+  {
+    store: usePolygonStore() as unknown as AnnotationToolStore<string>,
+    icon: 'mdi-pentagon-outline',
+  },
+];
```

Comment:
No action for now, but we will revisit this later on, potentially making a lookup table mapping the Tools enum to the stores.

Reply:
I like!